addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/handler/NestStructureHandler.java

```diff
@@ -104,17 +112,14 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     }
 
     @Override
-    public void onNewNestStructureData(Structure structure) {
-        if (isNotHandling(structure)) {
-            logger.debug("Structure {} is not handling update for {}", getStructureId(), structure.getStructureId());
-            return;
-        }
+    protected void update(Structure oldStructure, Structure structure) {
+        logger.debug("Updating {}", getThing().getUID());
 
-        logger.debug("Updating structure {}", structure.getStructureId());
+        updateLinkedChannels(oldStructure, structure);
 
-        setLastUpdate(structure);
-        updateChannels(structure);
-        updateStatus(ThingStatus.ONLINE);
+        if (!ThingStatus.ONLINE.equals(thing.getStatus())) {
```

Comment:
This could be an ==

Reply:
Good point. I'll change them to != so they are easier on the eyes.
You get so used to using .equals(..) everywhere that you forget they are not always necessary. ;-)